### ISSUE TITLE
`get_result` should return an error if multiple rows are returned

### ISSUE THREAD

**Issue by Diggsey** [diesel-rs/diesel#2446]
### Versions

All

## Problem Description

`RunQueryDsl` provides several methods for accessing query results:
- execute (returns number of rows affected)
- load/get_results (returns all rows)
- get_result (returns a single row)
- first (returns the first row)

I expected that `get_result` would return an error if multiple rows are returned (to differentiate it from `first`). However, it seems that it will just return the first row. Other ORMs ([SQLalchemy](https://docs.sqlalchemy.org/en/13/orm/exceptions.html?highlight=multipleresultsfound#sqlalchemy.orm.exc.MultipleResultsFound), [Django ORM](https://docs.djangoproject.com/en/3.0/ref/exceptions/#django.core.exceptions.MultipleObjectsReturned)) will raise an exception in this case.

This is dangerous: if code is using `get_result` then it clearly expects a single row to be returned, and if multiple rows are returned, it's a sign that the query may be wrong, or accidentally missing filters which should be present. If this expectation is invalid, then diesel should report an error sooner rather than later.

Unfortunately, this is a breaking change to fix. Personally I think that's still worth it due to the danger not fixing it poses. However, I would also be OK with the documentation being updated to call out this fact, and a new method being added to assert that a single item is returned.

**Comment by weiznich** [comment]
I'm not sure if I would really return an error in this case, is it can sometimes be the desired behavior (thought then you should probably just use `.first` in the first place as it appends `LIMIT 1` to your query.) Adding a note to the documentation sounds definitively like a good idea.
Theoretically we can also generate an error message for the case of an result set for more then one element there in the next version, as it will be diesel 2.0 anyway. So if someone adds some convincing arguments here I would accept a PR doing that.

**Comment by lsunsi** [comment]
I agree with @Diggsey and the main reason has to do with having an explicit intention. Having a method to make explicit the intention of wanting to get one and only one row from an update makes for clearer and stricter API while still supporting all use cases. Then .first, .get_result, .get_results all have different specific use cases.
